main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+func in(ctx context.Context, ch chan string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	var a string
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("Context cancelled, stop reading")
+			close(ch)
+			return
+		default:
+			fmt.Scan(&a)
+			ch <- a
+		}
+	}
+
+}
+
+func out(ctx context.Context, ch <-chan string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	f, _ := os.Create("file.txt")
+	defer f.Close()
+
+	for {
+		select {
+		case <-ctx.Done():
+			for v := range ch {
+				f.WriteString(v)
+			}
+			log.Println("Context cancelled, stopped writing")
+			return
+		}
```

Comment:
у тебя не записываются все введенные данные с консоли из-за вот этого места, ты начинаешь читать канал только после отмены контекста, а прочтется только одно значение (самое первое, которое положили в канал, а после канал закрывается по отмене контекста), здесь надо было использовать case селекта:
```suggestion
		case <-ctx.Done():
			log.Println("Context cancelled, stopped writing")
			return
		case v := <-ch:
			f.WriteString(v)
		}
```
в этом случае у тебя запишутся все введенные элементы (но кстати, в одной строке (!))

Reply:
короче, твоя программа работает в такой последовательности:
1) in прочитал с консоли первое значение, заблокировался, тк нет читающей горутины (потому что она не получила отмену контекста)
2) out получила отмену контекста - прочитала данные из канала - завершила работу
3) in разблокировалась, и отловила отмену контекста - завершила работу

таким образом другие введенные в консоль данные пропали